src/main/scala/mesosphere/marathon/core/instance/Instance.scala

```diff
@@ -70,8 +70,6 @@ case class Instance(
     * @return new instance in a provisioned state
     */
   def provisioned(agentInfo: Instance.AgentInfo, runSpec: RunSpec, tasks: Seq[Task], now: Timestamp): Instance = {
-    require(isScheduled, s"Instance '$instanceId' must not be in state '${state.condition}'. Scheduled instance is required to create provisioned instance.")
```

Comment:
I agree that `require(isScheduled...` case is not handled properly. Afaik, right now, it will lead to a crash/restart of _InstanceTrackerActor_ which is probably not what we want. However, handling this race properly requires a more work than just removing this line

Reply:
What kind of race condition do we have?